airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -327,7 +328,9 @@ def get_dag_runs(
     This endpoint allows specifying `~` as the dag_id to retrieve Dag Runs for all DAGs.
     """
     query = select(DagRun)
-
+    if dag_run_id_pattern:
+        query = query.filter(DagRun.run_id.like(dag_run_id_pattern))
+        
```

Comment:
looks like a good idea. But, can you consider making it similar to other filters and add it to filters in `paginated_select`?

Also, maybe you can add it to other dag run endpoints as well?
And, please add tests.

CC: @jason810496 , @bugraoz93 , @pierrejeambrun

Reply:
I agree, we are following that pattern. You can leverage `airflow.api_fastapi.common.parameters` where there are factories. You can build any filter you want easily even if it doesn't exist.

I am also concerned that we shouldn't accept any SQL query like this with accepting wildcards or so.
See https://labs.portcullis.co.uk/whitepapers/dos-attacks-using-sql-wildcards/. 
So no pattern from me if only string is supplied without patterns, that's okay.

Reply:
I agree, we are following that pattern. You can leverage `airflow.api_fastapi.common.parameters` where there are factories. You can build any filter you want easily even if it doesn't exist.

I am also concerned that we shouldn't accept any SQL query like this with accepting wildcards or so.
See https://labs.portcullis.co.uk/whitepapers/dos-attacks-using-sql-wildcards/. 
So no for patterns. If only string is supplied without patterns, that's okay.

Reply:
I also agree with adding unit tests. Could you please add them?